addOns/soap/src/main/java/org/zaproxy/zap/extension/soap/WSDLFilePassiveScanRule.java

```diff
@@ -57,8 +57,7 @@ public boolean isWsdl(HttpMessage msg) {
             HttpResponseHeader header = msg.getResponseHeader();
             String baseURL = msg.getRequestHeader().getURI().toString().trim();
             String contentType = header.getHeader(HttpHeader.CONTENT_TYPE).trim();
-            return baseURL.endsWith(".wsdl")
-                    || contentType.equals("text/xml")
+            return (baseURL.endsWith(".wsdl") && contentType.equals("text/xml"))
```

Comment:
Why just ending with `.wsdl` is no longer considered WSDL?

Reply:
That’s why I included two different groupings.

It can be any URL ending in wsdl with xml content type or wsdl/xml content type. Or, only URLs ending in wsdl with either content type.

Reply:
Or something I didn’t consider I suppose

Reply:
I think that the `"text/xml"` check can be omitted altogether and the check should just be
```java
baseURL.endsWith(".wsdl") || contentType.equals("application/wsdl+xml")
```

Reply:
Sounds good to me.

Reply:
> I think that the `"text/xml"` check can be omitted altogether and the check should just be
> 
> ```java
> baseURL.endsWith(".wsdl") || contentType.equals("application/wsdl+xml")
> ```

I added this and Now I think that `ShouldNotAlertWithJustXmlContentType` in unit test does not make any sense so should I remove it? 

In unit test I added three cases:

1. Should alert when file found with wsdl extension.
2. Should alert when wsdl content type found.
3. Should not alert when file extension is not wsdl.

Reply:
Yes, that test can be removed now.